ChaosMod/Effects/db/Peds/PedsSeriousKamikazes.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+	Effect by Moxi based on Killer Clowns by Last0xygen, explosives peds
+*/
+
+#include <stdafx.h>
+
+static std::list<Ped> kamikazeEnemies;
+static std::list<Object> bombObject;
+static int spawnTimer = -1;
+static Hash relationshipGroup;
+static int maxKamikazesToSpawn = 5;
+
+static Vector3 getRandomOffsetCoord(Vector3 startCoord, int minOffset, int maxOffset)
+{
+	Vector3 randomCoord;
+	randomCoord.z = startCoord.z;
+	float groundZ;
+	for (int i = 0; i < 10; i++) {
+		if (g_Random.GetRandomInt(0, 1) % 2 == 0)
+		{
+			randomCoord.x = startCoord.x + g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		else
+		{
+			randomCoord.x = startCoord.x - g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		if (g_Random.GetRandomInt(0, 1) % 2 == 0)
+		{
+			randomCoord.y = startCoord.y + g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		else
+		{
+			randomCoord.y = startCoord.y - g_Random.GetRandomInt(minOffset, maxOffset);
+		}
+		randomCoord.z = startCoord.z;
+		if (GET_GROUND_Z_FOR_3D_COORD(randomCoord.x, randomCoord.y, randomCoord.z, &groundZ, false, false))
+		{
+			randomCoord.z = groundZ;
+			break;
+		}
+	}
+	return randomCoord;
+}
+
+static void OnStop()
+{
+	for (Ped ped : kamikazeEnemies)
+	{
+		SET_PED_AS_NO_LONGER_NEEDED(&ped);
+	}
+	kamikazeEnemies.clear();
+
+	for (Object bomb : bombObject)
+	{
+		SET_OBJECT_AS_NO_LONGER_NEEDED(&bomb);
+	}
+	bombObject.clear();
+}
+
+static void OnStart()
+{
+	static const Hash playerGroup = GET_HASH_KEY("PLAYER");
+	ADD_RELATIONSHIP_GROUP("_HOSTILE_SERIOUS_KAMIKAZES", &relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, playerGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(5, playerGroup, relationshipGroup);
+	SET_RELATIONSHIP_BETWEEN_GROUPS(0, relationshipGroup, relationshipGroup);
+}
+
+static void OnTick()
+{
+	Ped playerPed = PLAYER_PED_ID();
+	Vector3 playerPos = GET_ENTITY_COORDS(playerPed, false);
+	int current_time = GET_GAME_TIMER();
+
+	for (std::list<Entity>::iterator it = kamikazeEnemies.begin(); it != kamikazeEnemies.end(); )
+	{
+		Ped kamikaze = *it;
+		Vector3 kamikazePos = GET_ENTITY_COORDS(kamikaze, false);
+		if (IS_PED_DEAD_OR_DYING(kamikaze, false) || IS_PED_INJURED(kamikaze) || GET_DISTANCE_BETWEEN_COORDS(playerPos.x, playerPos.y, playerPos.z, kamikazePos.x, kamikazePos.y, kamikazePos.z, false) > 100.f)
+		{
+			SET_ENTITY_HEALTH(kamikaze, 0, 0);
+			WAIT(300);
```

Comment:
No wait needed, this was copy pasted from my killer clowns which „animate“ their disappearance